pkg/core/basic_chain_test.go

```diff
@@ -596,3 +625,1634 @@ func TestBlockchain_StartFromExistingDB(t *testing.T) {
+			Scopes:  transaction.Global,
+		})
+		require.NoError(t, accs[0].SignTx(netmode.UnitTestNet, tx))
+		require.NoError(t, accs[3].SignTx(netmode.UnitTestNet, tx))
+		checkErr(t, core.ErrVerificationFailed, tx)
+	})
+	t.Run("OldTX", func(t *testing.T) {
+		tx := newTestTx(t, h, testScript)
+		require.NoError(t, accs[0].SignTx(netmode.UnitTestNet, tx))
+		e.AddNewBlock(t, tx)
+
+		err := bc.VerifyTx(tx)
+		require.True(t, errors.Is(err, core.ErrAlreadyExists))
+	})
+	t.Run("MemPooledTX", func(t *testing.T) {
+		tx := newTestTx(t, h, testScript)
+		require.NoError(t, accs[0].SignTx(netmode.UnitTestNet, tx))
+		require.NoError(t, bc.PoolTx(tx))
+
+		err := bc.PoolTx(tx)
+		require.True(t, errors.Is(err, core.ErrAlreadyExists))
+	})
+	t.Run("MemPoolOOM", func(t *testing.T) {
+		mp := mempool.New(1, 0, false)
+		tx1 := newTestTx(t, h, testScript)
+		tx1.NetworkFee += 10000 // Give it more priority.
+		require.NoError(t, accs[0].SignTx(netmode.UnitTestNet, tx1))
+		require.NoError(t, bc.PoolTx(tx1, mp))
+
+		tx2 := newTestTx(t, h, testScript)
+		require.NoError(t, accs[0].SignTx(netmode.UnitTestNet, tx2))
+		err := bc.PoolTx(tx2, mp)
+		require.True(t, errors.Is(err, core.ErrOOM))
+	})
+	t.Run("Attribute", func(t *testing.T) {
+		t.Run("InvalidHighPriority", func(t *testing.T) {
+			tx := newTestTx(t, h, testScript)
+			tx.Attributes = append(tx.Attributes, transaction.Attribute{Type: transaction.HighPriority})
+			require.NoError(t, accs[0].SignTx(netmode.UnitTestNet, tx))
+			checkErr(t, core.ErrInvalidAttribute, tx)
+		})
+		t.Run("ValidHighPriority", func(t *testing.T) {
+			tx := newTestTx(t, h, testScript)
+			tx.Attributes = append(tx.Attributes, transaction.Attribute{Type: transaction.HighPriority})
+			tx.NetworkFee += 4_000_000 // multisig check
+			tx.Signers = []transaction.Signer{{
+				Account: committee.ScriptHash(),
+				Scopes:  transaction.None,
+			}}
+			rawScript := committee.Script()
+			size := io.GetVarSize(tx)
+			netFee, sizeDelta := fee.Calculate(bc.GetBaseExecFee(), rawScript)
+			tx.NetworkFee += netFee
+			tx.NetworkFee += int64(size+sizeDelta) * bc.FeePerByte()
+			tx.Scripts = []transaction.Witness{{
+				InvocationScript:   committee.SignHashable(uint32(netmode.UnitTestNet), tx),
+				VerificationScript: rawScript,
+			}}
+			require.NoError(t, bc.VerifyTx(tx))
+		})
+		/*
+			// TODO: unexported API usage?
+				t.Run("Oracle", func(t *testing.T) {
+					orc := bc.contracts.Oracle
+					req := &state.OracleRequest{GasForResponse: 1000_0000}
+					require.NoError(t, orc.PutRequestInternal(1, req, bc.dao))
+
+					oracleScript, err := smartcontract.CreateMajorityMultiSigRedeemScript(oraclePubs)
+					require.NoError(t, err)
+					oracleHash := hash.Hash160(oracleScript)
+
+					// We need to create new transaction,
+					// because hashes are cached after signing.
+					getOracleTx := func(t *testing.T) *transaction.Transaction {
+						tx := bc.newTestTx(h, orc.GetOracleResponseScript())
```

Comment:
This might be easily moved into function from method.

Reply:
Sure, fixed.